pkgs/by-name/dp/dprint/plugins/update-plugins.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i python -p nix 'python3.withPackages (pp: with pp; [ requests ])'
+
+import json
+import subprocess
+import requests
+
+
+def nix_prefetch_url(url, algo="sha256"):
+    hash = (
+        subprocess.check_output(["nix-prefetch-url", "--type", algo, url])
+        .decode("utf-8")
+        .rstrip()
+    )
+    sri = (
+        subprocess.check_output(
+            ["nix", "hash", "convert", "--hash-algo", algo, "--to", "sri", hash]
+        )
+        .decode("utf-8")
+        .rstrip()
+    )
+    return sri
+
+
+def update_plugins():
+    data = requests.get("https://plugins.dprint.dev/info.json").json()["latest"]
+    with open("plugins.json", "w+", encoding="utf8") as f:
+        result = {}
+        for e in data:
+            # using configKey instead of name here
+            # name can have slashes for eg. g-plane/malva
+            # but configKey is a good sensible unique word per plugin
+            result[e["configKey"]] = {
+                "url": e["url"],
+                "hash": nix_prefetch_url(e["url"]),
+            }
+        json.dump(result, f, indent=2)
+        f.write("\n")
+
+
+update_plugins()
```

Comment:
One thing I'm not sure of is if the update framework is able to call this file. It might have to be attached as an passthru.updateScript to each plugin.

Reply:
I will add passthru.updateScript as the updatescript, maybe adapt it slightly to accept a plugin name as an argument, and update that single plugin.

I don't know if that is possible, passing arguments to the updateScript.

Reply:
used `UPDATE_NIX_PNAME`, allows individual updates.